src/rgw/rgw_tracer.h

```diff
@@ -0,0 +1,77 @@
+#ifndef TRACER_WRAPPER_H
+#define TRACER_WRAPPER_H
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+#include <jaegertracing/Tracer.h>
+#include <yaml-cpp/yaml.h>
+#include "common/config.h"
+#include "common/ceph_context.h"
+
+
+namespace RGW_Tracing
+{
+    class TracerConfiguration : public md_config_obs_t {
+        private:
+            bool enabled = true;
+
+        public:
+            TracerConfiguration();
+            bool is_enabled() const;
+            const char** get_tracked_conf_keys() const override;
+            void handle_conf_change(const ConfigProxy& conf,
+                                    const std::set<std::string>& changed) override;
+    };
+
+    
+
+    class Span {
+        private:
+            std::unique_ptr<opentracing::Span> span;
+        public:
+            Span(std::unique_ptr<opentracing::Span> _span);
+            ~Span();
+
+            std::string get_baggage(std::string key) const;
+            opentracing::SpanReference get_reference() const;
+        
+            void set_tag(std::string key,std::string value);
+            void set_log(std::string key,std::string value);
+            /* set <key,value> pair which will be propagated to child spans */
+            void set_baggage(std::string key,std::string value);
+            
+    };
+
+
+    class Tracer {
+    private:
+        std::shared_ptr<opentracing::Tracer> open_tracer;
+    
+    public:
+        const static std::shared_ptr<opentracing::Tracer>   noop_tracer;
+        static TracerConfiguration    tracer_conf;
+
+
+        Tracer(jaegertracing::Config& conf);
+        Tracer( std::string& service_name,
+                const jaegertracing::Config& conf,
+                const std::shared_ptr<jaegertracing::logging::Logger>& logger);
+        Tracer(std::string configFilePath);
+        ~Tracer();
+        bool is_enabled() const;
+        std::unique_ptr<Span> start_trace(std::string trace_name);
+        std::unique_ptr<Span> start_span(std::string span_name,
+                                                      std::unique_ptr<RGW_Tracing::Span>& parentSpan);
+        
+
+        
+    };
+
+
+}
+
+namespace Jaeger_Configuration{
+    extern jaegertracing::Config& get_jaeger_default_configuration();
```

Comment:
why do we need this?

Reply:
it's just a forward declaration. the function itself is not so necessary since it just returns a global configuration object.
we can just use the global config object.